profiles/api/api.py

```diff
@@ -16,10 +16,20 @@
 DEFAULT_PER_PAGE = 20
 
 
+def cache_control_headers(func):
+    def wrapper(*args, **kwargs):
+        response = func(*args, **kwargs)
+        response.headers['Cache-Control'] = 'max-age=300, public, stale-if-error=86400,' \
+                                            'stale-while-revalidate=300'
+        return response
+    return wrapper
+
+
 def create_blueprint(profiles: Profiles) -> Blueprint:
     blueprint = Blueprint('api', __name__)
 
-    @blueprint.route('/profiles')
+    @blueprint.route('/profiles', endpoint='_list')
+    @cache_control_headers
```

Comment:
That's what I was aiming for, but with an optional argument to say don't cache (for the OAuth routes).

Following advice on t'interweb was trying things like wrapt, but could never get it to work. Looks like that was never needed...

(Or was adding `endpoint` the fix?)

Reply:
The decorator effectively scraps the function name in a way, so the `@blueprint.route` was complaining as it thought you (we) was trying to register the same function for multiple routes. Defining the name in the `endpoint` arg sorted that, but I have just cleaned it up further by using `@wraps` on the internal wrapper function, this in turn keeps the name etc safe, removing the need for the `endpoint` arg.

You can add args to the decorators but it needs amending to handle that. Show me a sudo code sample and I can do it if you like.